test/common/http/match_delegate/config_test.cc

```diff
@@ -202,9 +239,8 @@ TEST(MatchWrapper, WithNoMatcher) {
 )EOF");
 
   MatchDelegateConfig match_delegate_config;
-  EXPECT_THROW_WITH_REGEX(
-      match_delegate_config.createFilterFactoryFromProto(config, "", factory_context),
-      EnvoyException, "one of `matcher` and `matcher_tree` must be set.");
+  auto cb = match_delegate_config.createFilterFactoryFromProto(config, "", factory_context);
+  EXPECT_TRUE(cb);
```

Comment:
This doesn't verify the skip behavior.

We can verify it something like [here](https://github.com/tyxia/envoy/blob/ea38693125f22f6374328023c088ed602e33874d/test/common/http/match_delegate/config_test.cc#L333): verify the those decode function has been called Times(0) and FilterStatus is continue.

Reply:
I think your `WithNoMatcher` test in most recent revision looks good to me.